.github/workflows/ci.yml

```diff
@@ -3,9 +3,24 @@
 name: WildFly Quickstarts CI
 
 on:
+  push:
+    branches:
+      - 'main'
   pull_request:
     types: [opened, synchronize, reopened, ready_for_review]
+
+# Only run the latest job
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
 jobs:
+  wildfly-build:
```

Comment:
The workflow should be "branch abstract", to work on the NN.x branches without any modifications. This is why all dependency checkouts use ref: ${{ github.base_ref }}

Reply:
To be clear, the "with deps" job of the 29.x branch of QS should build with WFLY 29.x and BOMs 29.x built with such WFLY.

Reply:
Yes, we could change the `${{ github_base_ref}}` here for sure. I'll fix that.